### PR TITLE
Implement intrinsics::write_bytes

### PR DESCRIPTION
**Pull request by hermanventer**
## Description

Implement intrinsics::write_bytes.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage
- [ ] Code cleanup or just keeping up with the latest Rustc nightly

## How Has This Been Tested?
./validate.sh
ran MIRAI over Libra

### CODE REVIEW

File: checker/src/call_visitor.rs [hunk 1]
```diff
@@ -1754,6 +1758,92 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
...
+            }
+            source_value
+        }
+        if elem_type.is_primitive() {
+            let dest_pattern = Path::new_slice(dest_path, count_value);
+            let source_value = repeated_bytes(elem_size, byte_value);
+            self.block_visitor
+                .bv
+                .current_environment
+                .update_value_at(dest_pattern, source_value);
```
Comment (stonebuddha) [inline checker/src/call_visitor.rs:1794]: Is it supposed to be a `copy_or_move_elements` call, because the target path is a pattern?

Comment (hermanventer) [reply checker/src/call_visitor.rs:1794]: No, we don't want expansion of the pattern at this point since the caller is going to do that anyway.

Comment (stonebuddha) [reply checker/src/call_visitor.rs:1794]: Ah, the caller will do that during refinement?

Comment (hermanventer) [reply checker/src/call_visitor.rs:1794]: The caller's caller, yes.

**Pull request merged**